src/messages/TextMessage.vue

```diff
@@ -1,37 +1,47 @@
 <template>
-  <div class="sc-message--text" :style="messageColors">
+  <div class="sc-message--text" v-bind:class="{ 'confirm-delete' : deleteConfirmation == true }" :style="messageColors">
     <template>
       <div class="sc-message--toolbox" :style="{background: messageColors.backgroundColor}">
-        <button
-          v-if="showEdition && me && message.id != null && message.id != undefined"
-          :disabled="isEditing"
-          @click="edit"
-        >
-          <IconBase :color="isEditing ? 'black' : messageColors.color" width="10" icon-name="edit">
-            <IconEdit />
-          </IconBase>
+        <button v-if="showEdition && me && message.id != null && message.id != undefined" @click="edit" :disabled="isEditing">
+          <icon-base :color="isEditing? 'black': messageColors.color" width="10" icon-name="edit">
+            <icon-edit />
+          </icon-base>
         </button>
-        <button
-          v-if="showDeletion && me && message.id != null && message.id != undefined"
-          @click="$emit('remove')"
-        >
-          <IconBase :color="messageColors.color" width="10" icon-name="remove">
-            <IconCross />
-          </IconBase>
-        </button>
-        <slot name="text-message-toolbox" :message="message" :me="me"> </slot>
+        <div v-if="showDeletion">
+          <div v-if="showConfirmationDeletion">
+            <div v-if="deleteConfirmation">
```

Comment:
It may help to document what these flags stand for, or rename them to be a bit more specific. Could also merge related state into a computed object and use that (and provide code-comments there for help).